### PR TITLE
Tools: fix fpgainfo mac command to prints  mac count and address list.

### PR DESCRIPTION
**Pull request by anandaravuri**
Print mac info function reads mac address and mac count from sysfs and prints mac info

 cat  /sys/class/fpga_region/region0/dfl-fme.0/dfl_dev.2/mac_count  8
 cat  /sys/class/fpga_region/region0/dfl-fme.0/dfl_dev.2/mac_address  78:56:34:12:ab:90

 sudo ./bin/fpgainfo mac
  FPGA SmartNIC N6010 Card
  Board Management Controller, MAX10 NIOS FW version: 0.2.0
  Board Management Controller, MAX10 Build version: 0.2.0
  //****** MAC ******//
  Object Id                        : 0xEF00000
  PCIe s:b:d.f                     : 0000:B1:00.0
  Device Id                        : 0xBCCE
  Socket Id                        : 0x00
  Ports Num                        : -527316905
  Bitstream Id                     : 0x10005
  Bitstream Version                : 0.0.0
  Pr Interface Id                  : 0e7f3e69-0ca3-5a17-858c-8bd1e8ac621e
  Number of MACs                   : 8
  MAC address 0                    : 78:56:34:12:AB:90
  MAC address 1                    : 78:56:34:12:AB:91
  MAC address 2                    : 78:56:34:12:AB:92
  MAC address 3                    : 78:56:34:12:AB:93
  MAC address 4                    : 78:56:34:12:AB:94
  MAC address 5                    : 78:56:34:12:AB:95
  MAC address 6                    : 78:56:34:12:AB:96
  MAC address 7                    : 78:56:34:12:AB:97

Signed-off-by: anandaravuri <ananda.ravuri@intel.com>

### CODE REVIEW

File: tools/libboard/board_n6010/board_n6010.h [hunk 1]
```diff
@@ -34,6 +34,13 @@ extern "C" {
 #endif /* __cplusplus */
 
 #define SYSFS_MAX_SIZE             256
+#define MAC_BYTE_SIZE              4
+#define MAC_BUF_LEN                19
+
+typedef union _cvl_mac {
+	unsigned int dword;
+	unsigned char byte[MAC_BYTE_SIZE];
+} cvl_mac;
```
Comment (fpgamatt) [inline tools/libboard/board_n6010/board_n6010.h:43]: What is the cvl prefix in the name for?

File: tools/libboard/board_n6010/board_n6010.h [hunk 2]
```diff
@@ -34,6 +34,13 @@ extern "C" {
 #endif /* __cplusplus */
 
 #define SYSFS_MAX_SIZE             256
+#define MAC_BYTE_SIZE              4
+#define MAC_BUF_LEN                19
+
+typedef union _cvl_mac {
+	unsigned int dword;
+	unsigned char byte[MAC_BYTE_SIZE];
```
Comment (fpgamatt) [inline tools/libboard/board_n6010/board_n6010.h:42]: ethernet addresses are typically defined as 6 byte quantities.  Why not do the normal approach?

File: tools/libboard/board_n6010/board_n6010.c [hunk 3]
```diff
@@ -257,83 +257,63 @@ fpga_result read_max10fw_version(fpga_token token, char *max10fw_ver, size_t len
...
+
+	res = read_sysfs(token, DFL_SYSFS_MACADDR_PATH, (char *)buf, MAC_BUF_LEN - 1);
 	if (res != FPGA_OK) {
-		OPAE_ERR("Failed to get read object");
+		OPAE_ERR("Failed to read mac information");
 		return res;
 	}
 
-	ether_aton_r(mac_buf, mac_addr);
+	sscanf(buf, "%x:%x:%x:%x:%x:%x", &mac_byte[0], &mac_byte[1],
```
Comment (fpgamatt) [inline tools/libboard/board_n6010/board_n6010.c:278]: ether_aton_r is a better choice than sscanf.

File: tools/libboard/board_n6010/board_n6010.h (hunk 1 above)
Comment (anandaravuri) [reply tools/libboard/board_n6010/board_n6010.h]: removed struct cvl_mac

File: tools/libboard/board_n6010/board_n6010.h (hunk 2 above)
Comment (anandaravuri) [reply tools/libboard/board_n6010/board_n6010.h]: fixed ,removed struct cvl_mac

File: tools/libboard/board_n6010/board_n6010.c (hunk 3 above)
Comment (anandaravuri) [reply tools/libboard/board_n6010/board_n6010.c]: fixed ether_aton_r  function.

**Review by fpgamatt — APPROVED** [review]

**Review by tswhison — APPROVED** [review]

**Review by michelleyho — APPROVED** [review]

**Pull request merged**